website/mails/mail_callbacks.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+
+from website import settings
+from modularodm import Q
+from modularodm.exceptions import NoResultsFound, MultipleResultsFound
+
+def no_addon(email):
+    if len(email.user.get_addons()) is 0 and email.user.is_registered:
+        return True
+
+def no_login(email):
+    if not email.user.is_registered or email.user.date_last_login > datetime.utcnow() - settings.NO_LOGIN_WAIT_TIME:
+        return False
+    return True
+
+def new_public_project(email):
+    from website.models import Node
+    node = Node.find_one(Q('_id', 'eq', email.data['nid']))
+    public = email.find_same_sent()
+    if node.is_public and not len(public):
+        return True
+    return False
+
+def welcome_osf4m(email):
+    from website.addons.osfstorage.model import OsfStorageFileNode
```

Comment:
If this is to resolve a circular import comment that please otherwise move it up top.